manager/integration/tests/test_migration.py

```diff
@@ -147,7 +147,40 @@ def test_migration_with_failed_replica(clients, volume_name):  # NOQA
        And the old failed replica will be cleaned up.
     8. Validate initially written test data.
     """
-    return
+    client, volume, data = setup_migration_test(clients,
+                                                volume_name,
+                                                replica_cnt=3)
+
+    current_node = common.get_self_host_id()
+    host1, host2 = get_hosts_for_migration_test(clients)
+    volume.attach(hostId=current_node)
+    volume = common.wait_for_volume_healthy(client, volume_name)
+
+    exec_cmd = ["rm", "-rf",  "/var/lib/longhorn/replicas"]
+    subprocess.check_output(exec_cmd)
+    volume = common.wait_for_volume_degraded(client, volume_name)
+
+    volume.attach(hostId=host1)
+
+    volume = common.wait_for_volume_migration_ready(client, volume_name)
+
+    volume.detach(hostId=current_node)
+
+    volume = common.wait_for_volume_migration_node(client,
+                                                   volume_name,
+                                                   host1)
```

Comment:
Need to verify the old failed replica is removed.

Reply:
This comment is missed

Reply:
Failed replica will keep exist since the replica count was 3 in my test environment, so I did not change code for this comment

Reply:
IIRC, after the detachment, all failed replicas will be removed and new replicas will be replenished.